test/AutoDiff/SILOptimizer/pullback_generation.swift

```diff
@@ -0,0 +1,129 @@
+// Pullback generation tests written in Swift
+
+// RUN: %target-swift-frontend -emit-sil -verify  -Xllvm -debug-only=differentiation %s  2>&1 | %FileCheck %s
+
+import _Differentiation
+
+//===----------------------------------------------------------------------===//
+// Pullback generation - `struct_extract`
+// - Operand is piecewise materializable
+//===----------------------------------------------------------------------===//
+struct PiecewiseMaterializable: Differentiable, Equatable, AdditiveArithmetic {
+    public typealias TangentVector = Self
+    
+    var a: Float
+    var b: Double
+}
+
+@differentiable(reverse)
+func f1(v: PiecewiseMaterializable) -> Float {
+    v.a
+}
+
+// CHECK-LABEL: [ORIG]  %2 = struct_extract %0 : $PiecewiseMaterializable, #PiecewiseMaterializable.a // user: %3
```

Comment:
Maybe it will make sense to check the generated SIL and not the debug output? After all, you can do `-Xllvm -sil-print-after=differentiation`

Reply:
Yeah sure.